internal/hardware/validator.go

```diff
@@ -244,8 +285,20 @@ func (v *validator) getOCPHostRoleRequirementsForVersion(cluster *common.Cluster
 	return *requirements.WorkerRequirements, nil
 }
 
+func (v *validator) getOCPInfraEnvHostRoleRequirementsForVersion(infraEnv *common.InfraEnv, role models.HostRole) (models.ClusterHostRequirementsDetails, error) {
+	requirements, err := v.getOCPRequirementsForVersion(infraEnv.OpenshiftVersion)
+	if err != nil {
+		return models.ClusterHostRequirementsDetails{}, err
+	}
+
+	if role == models.HostRoleMaster {
+		return *requirements.MasterRequirements, nil
+	}
+	return *requirements.WorkerRequirements, nil
```

Comment:
I know that today if it's not a master it is a worker but I think it will be nicer to have 'else worker' as more structured and maybe in the future it won't be the only two options

Reply:
actually in such case golang does not like if /else

Reply:
it will if after the if/else you will return an error which is what is expected if you get something that is not master/worker

Reply:
fixed